packages/react/src/analytics/analytics.ts

```diff
@@ -56,6 +66,47 @@ class AnalyticsWeb extends Analytics {
     }
   }
 
+  override async getTrackEventData(
+    type: TrackTypesValues,
+    event: string,
+    properties?: EventProperties,
+    eventContext?: ContextData,
+  ): Promise<EventData<TrackTypesValues>> {
+    this.processContext(eventContext);
+
+    return await super.getTrackEventData(type, event, properties, eventContext);
+  }
+
+  override context(): Promise<ContextData>;
+  override context(key: string): Promise<unknown>;
+  override async context(key?: string) {
+    if (key && key !== 'library.version') {
+      return super.context(key);
+    }
+
+    const context = await super.context();
+
+    this.processContext(context);
+
+    // if key are set and goes to this path, then key must be equal to analyticsVersion
+    return key ? get(context, key) : context;
+  }
+
+  processContext(context?: ContextData) {
+    if (context?.library) {
```

Comment:
Context and library will always be defined, so this check is not necessary.

Reply:
Following the code, this method are created to apply same code in two different places, avoiding duplication.

One of the calls, are on getTrackEventData, which context parameter could be undefined, so In this method I need to check if context are filled.

But I could remove check of library, because if context are assigned, by contract, it requires library property are set too.